oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -849,18 +834,21 @@ impl Runtime {
                 })?;
             }
             let statuses = self.readers_statuses(node_id, &readers);
+            // Transcribe the status for valid channels back to the original position
+            // in the list of all statuses.
+            for i in 0..readers.len() {
+                all_statuses[reader_pos[i]] = statuses[i];
+            }
 
-            let all_unreadable = statuses.iter().all(|&s| {
-                s == ChannelReadStatus::InvalidChannel || s == ChannelReadStatus::Orphaned
+            let any_invalid = statuses.iter().any(|&s| {
```

Comment:
Given the newly updated documentation, I would expect simply that if any reader status is different from NotReady, then the entire vector is returned immediately. Or did I misunderstand?

Reply:
Yes. That is the new logic. So, it waits only if all statuses are `NotReady`. I could change the condition to check that instead. It would be clearer in my opinion.